### ISSUE TITLE
Failed to alert ConsecutiveAppendsShouldReuse wrapped with block.

### ISSUE THREAD

**Issue by tjuwhy** [pmd/pmd#3775]
**Affects PMD Version: 6.42.0** 

**Rule:**  
ConsecutiveAppendsShouldReuse

Please provide the rule name and a link to the rule documentation:  https://pmd.github.io/pmd-6.42.0/pmd_rules_java_performance.html#consecutiveappendsshouldreuse

**Description:**  failed to detect the consecutive appends wrapped with block

**Code Sample demonstrating the issue:**

```java
public class ConsecutiveAppend {
    public static void main(String[] args) {
        StringBuilder builder = new StringBuilder();
        { builder.append("1"); } // false negative here
        builder.append("1"); 

        System.out.println(builder);
    } 
} 
```
**Expected outcome:**

PMD should report a violation at line 4, but doesn't. This is a false-negative.

**Running PMD through:** *CLI*
